### PR TITLE
Bugfixes cleanup

### PR DESCRIPTION
**Pull request by IanAtCredo**
## Describe your changes
Few changes:

* Refactored process_metrics. `Process_metrics` has been removed from fairness and performance and added to `metrics`. The routing of different kinds of metrics to different metric frames has been incorporated into the `setup_metric_frames` functionality.
* Added sensitive_feature labeling to the base `evaluator`. Any evaluator that takes a sensitive_feature will automatically label all evidence created with the appropriate labels, assuming that `get_info` (renamed from `get_container_info`) has been added.
* 
## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have built basic tests for new functionality (particularly new evaluators)
- [ ] If new libraries have been added, I have checked that [readthedocs](https://readthedocs.org/projects/credoai-lens/) API documentation is constructed correctly
- [ ] Will this be part of a major product update? If yes, please write one phrase about this update.

## Extra-mile Checklist
- [ ] I have thought expansively about edge cases and written tests for them

### CODE REVIEW

File: credoai/evaluators/utils/fairlearn.py [hunk 1]
```diff
@@ -16,47 +17,80 @@ def create_metric_frame(metrics, y_pred, y_true, sensitive_features):
     )
 
 
+def filter_processed_metrics(
+    processed_metrics, metric_categories=None, Xmetric_categories=None, takes_prob=None
+):
+    """Filters processed metrics
```
Comment (fabrizio-credo) [inline credoai/evaluators/utils/fairlearn.py:23]: As part of the standardization of the docstrings, I had chosen to put the title in a new line:

```
"""
Filters processed metrics
```
This can be a setting in VSCode formatter, so we don't have to remember doing it.

It is just a stylistic choice in general, the only place where it matters is in evaluators docstrings, since they are parsed to create documentation on RTD.

**Review by fabrizio-credo — APPROVED** [review]
Looks good to me, Fairness/Performance look so much tidier now.
